wandb/wandb_run.py

```diff
@@ -446,13 +461,48 @@ def use_artifact(self, artifact, type=None, aliases=None):
                         'aliases': aliases
                     }
                 })
+                return artifact
             elif isinstance(artifact, ApiArtifact):
                 self.api.use_artifact(artifact.id)
                 return artifact
             else:
                 raise ValueError('You must pass an artifact name (e.g. "pedestrian-dataset:v1"), an instance of wandb.Artifact, or wandb.Api().artifact() to use_artifact')
 
-    def log_artifact(self, artifact, aliases=['latest']):
+    def log_artifact(self, artifact_or_path, name=None, type=None, aliases=['latest']):
+        """ Declare an artifact as output of a run.
+
+        Args:
+            artifact_or_path (str or Artifact): A path to the contents of this artifact,
+                can be in the following forms:
+                    /local/directory
+                    /local/directory/file.txt
+                    s3://bucket/path
+                You can also pass an Artifact object created by calling `wandb.Artifact`
+            name (str, optional): An artifact name. May be prefixed with entity/project. Valid names
+                can be in the following forms:
+                    sequence_name:version
+                    sequence_name:alias
+                    digest
+                this will default to the basename of the path if not specified
+            type (str): The type of artifact to log, examples include "dataset", "model"
+            aliases (list, optional): Aliases to apply to this artifact, defaults to ["latest"]
+        Returns:
+            A :obj:`Artifact` object.
+        """
+        if isinstance(artifact_or_path, six.string_types):
+            if name is None:
+                name = "run-%s-%s" % (self.id, os.path.basename(artifact_or_path))
```

Comment:
Is this the behavior we want? This means that different runs adding the same path will produce different artifact collections everytime. This also seems to be out of sync with the comment above: `this will default to the basename of the path if not specified`.

Reply:
I'll change the doc string.  @shawnlewis can you confirm this is the behaviour you want?

Reply:
Yes this is what we want for now.